src/app/shared/utils/ck-upload-adapter.ts

```diff
@@ -0,0 +1,83 @@
+import { DocumentService } from 'app/services/document.service';
+
+// This utility function is a template directly from CK,
+// modified to use our existing DocumentService.
+// https://ckeditor.com/docs/ckeditor5/latest/framework/guides/deep-dive/upload-adapter.html
+
+export class CkUploadAdapter {
+  private imageURL: string;
+  private imageUploadData: FormData;
+
+  constructor(
+    private loader,
```

Comment:
```suggestion
    private loader: FileLoader,
```

See: https://ckeditor.com/docs/ckeditor5/latest/api/module_upload_filerepository-FileLoader.html

Reply:
Since we are using a custom instance of CK Editor, I am unable to find the FileLoader library and am very hesitant to start installing various CKEditor libraries that may or may not work with our instance...